lib/client.js

```diff
@@ -53,6 +54,7 @@ module.exports = class Client extends EventEmitter {
     if (!('port' in opts)) opts.port = this.port
     if (!opts.expectStatusCode) opts.expectStatusCode = 200
     if (!opts.protocol) opts.protocol = this.protocol
+    if (!opts.headers) opts.headers = this.headers
```

Comment:
Need to clone `this.headers` because the object will be mutated below.

```suggestion
    if (!opts.headers) opts.headers = { ...this.headers }
```